### PR TITLE
Copy Site: Include Marketplace Plugins when purchasing a new Copy Site

### PR DESCRIPTION
**Pull request by danielbachhuber**
See https://github.com/Automattic/dotcom-forge/issues/1514

## Proposed Changes

Includes any Marketplace plugins on the origin site in the copy site checkout flow.

## Testing Instructions

1. Create a new Business site and take it Atomic.
2. Purchase WooCommerce Subscriptions.
3. Initiate 'Copy site' flow.
4. Verify WooCommerce Subscriptions is included in the checkout cart.
5. Purchase the site copy.
6. Verify WooCommerce Subscriptions add-on is included on your new site.

### CODE REVIEW

File: client/sites-dashboard/components/sites-ellipsis-menu.tsx [hunk 1]
```diff
@@ -204,6 +209,11 @@ const CopySiteItem = ( { recordTracks, site }: SitesMenuItemProps ) => {
 			onClick={ () => {
 				clearSignupDestinationCookie();
 				setPlanCartItem( { product_slug: plan.product_slug } );
+				purchases.map( ( purchase ) => {
+					if ( 'marketplace_plugin' === purchase.productType ) {
+						// setPlanCartItem( { product_slug: purchase.productSlug } );
```
Comment (danielbachhuber) [inline client/sites-dashboard/components/sites-ellipsis-menu.tsx]: Not quite sure what's appropriate here, so I've asked shillings: p1674064554074069-slack-C0117V2PCAE

Comment (danielbachhuber) [reply client/sites-dashboard/components/sites-ellipsis-menu.tsx]: I managed to hack a product into the cart! https://github.com/Automattic/wp-calypso/pull/72291/commits/258d7cf63f4b5a003b7024be3371b3d5358b00d9#diff-3a158217b3c977b514f4fe1c284c0fd2e8072619c0c29a80e7b7b7bee8bc3623R115-R117

**Comment by danielbachhuber** [comment]
@kozer If you'd like to work on this tomorrow, feel free to assign yourself to https://github.com/Automattic/dotcom-forge/issues/1514 too and submit pull requests against this PR.

The first task will be to update `<CopySiteItem>` so it properly writes any product cart data to the Stepper store. `SiteCreationStep` then needs to be updated to read from the Stepper store and properly add the products to the cart.

Once you've done that, you'll notice checking out doesn't actually work 🙃 

CleanShot 2023-01-18 at 13 39 16@2x

Need to look into what should happen there...

**Comment by kozer** [comment]
I bring in `use-copy-site` hook in, and did the appropriate changes in commits https://github.com/Automattic/wp-calypso/pull/72291/commits/08297019aeaba2a17167d75eed0eab032f2cf387 and https://github.com/Automattic/wp-calypso/pull/72291/commits/4d65ddd297cdc0fe74a7155559b155fbb1a265ab

cc: @wojtekn , @danielbachhuber @sejas

**Pull request merged**